scripts/builtin/xgboostPredict.dml

```diff
@@ -0,0 +1,147 @@
+# INPUT PARAMETERS:
+# ----------------------------------------------------------------------------
+# NAME                              TYPE      DEFAULT  MEANING
+# ----------------------------------------------------------------------------
+# X                                 Matrix    ---      Matrix of feature vectors we want to predict (X_test)
+# M                                 Matrix    ---      The model created at xgboost
+# sml_type                          Integer   1        Supervised machine learning type: 1 = Regression(default), 2 = Classification
+# learning_rate                     Double    0.3     the learning rate used in the model
+
+# RETURN VALUES
+# ----------------------------------------------------------------------------
+# NAME     TYPE      DEFAULT  MEANING
+# ----------------------------------------------------------------------------
+# P        Matrix    ---     The predictions of the samples using the given xgboost model. (y_prediction)
+# ----------------------------------------------------------------------------
+
+m_xgboostPredict = function(Matrix[Double] X, Matrix[Double] M, Integer sml_type = 1, Double learning_rate = 0.3
```

Comment:
is a learning rate needed for predictions?

Reply:
learning rate is needed for prediction:   
init_predition + learning_rate * OutputTree1 + learning_rate * OutputTree2 + .... + learning_rate * OutputTreeN
Alternativly we can add the learning rate at the M matrix (eg at init prediction col), but it does not have a real with the M matrix